File: 프로그래머스/lv1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.py
def solution(n):
    answer = 0
    num_list = []
    
    # n을 각 자릿수로 분리 
    for i in str(n):
        num_list += [i]
        
    # 큰 수부터 정렬
    num_list.sort(reverse=True)
    
    # 정렬한 리스트 문자열로 변환
    n = ''
    for i in num_list:
        n += i
    
    # 문자열을 숫자로 변환
    answer = int(n)
    
    return answer

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.java
class Solution {
    public long solution(long n) {
        long answer = 0;
        
        int[] cntList = new int[10];
        while (n > 0) {
            int i = (int)(n % 10);
            n /= 10;
            cntList[i]++;
        }
        
        String s = "";
        
        for (int i = 9; i > -1; i--) {
            while (cntList[i] > 0) {
                s += i + "";
                cntList[i]--;
            }
        }
        
        answer = Long.parseLong(s);
        
        
        return answer;
    }
}